### ISSUE TITLE
[android] crash when exiting usb backend when weak_authority=1

### ISSUE THREAD

**Issue by starryalley** [libusb/libusb#841]
In android I have to set weak authority:
`libusb_set_option(NULL, LIBUSB_OPTION_WEAK_AUTHORITY);`

When unplugging the USB device, the app crashes in the exit callback:
```
2021-01-10 17:01:35.918 22547-22547/[myapp here] A/libc: path/to/libusb/os/linux_usbfs.c:424: void op_exit(struct libusb_context *): assertion "init_count != 0" failed
```

We should check weak_authority and skip doing the cleanup in op_exit.

**Comment by mcuee** [comment]
This looks correct to me.

**Comment by dickens** [comment]
Fixed in e78cafb
